storage/client/src/main/java/com/azure/storage/file/FileAsyncClient.java

```diff
@@ -3,77 +3,392 @@
-import com.azure.storage.file.models.FileRangeInfo;
+import com.azure.storage.file.models.FileRange;
 import com.azure.storage.file.models.FileRangeWriteType;
 import com.azure.storage.file.models.FileUploadInfo;
+import com.azure.storage.file.models.FileUploadRangeHeaders;
+import com.azure.storage.file.models.FilesCreateResponse;
+import com.azure.storage.file.models.FilesDownloadResponse;
+import com.azure.storage.file.models.FilesForceCloseHandlesResponse;
+import com.azure.storage.file.models.FilesGetPropertiesResponse;
+import com.azure.storage.file.models.FilesGetRangeListResponse;
+import com.azure.storage.file.models.FilesListHandlesResponse;
+import com.azure.storage.file.models.FilesSetHTTPHeadersResponse;
+import com.azure.storage.file.models.FilesSetMetadataResponse;
+import com.azure.storage.file.models.FilesStartCopyResponse;
+import com.azure.storage.file.models.FilesUploadRangeResponse;
 import com.azure.storage.file.models.HandleItem;
 import io.netty.buffer.ByteBuf;
+import java.net.URL;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.reactivestreams.Publisher;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
-
 import java.util.Map;
 
 public class FileAsyncClient {
-    FileAsyncClient() {
-        throw new UnsupportedOperationException();
+    private final AzureFileStorageImpl azureFileStorageClient;
+    private final String shareName;
+    private final String filePath;
+    private final String shareSnapshot;
+
+    /**
+     * Constructor of FileAsyncClient
+     * @param azureFileStorageClient
+     * @param shareName
+     * @param filePath
+     */
+    FileAsyncClient(AzureFileStorageImpl azureFileStorageClient, String shareName, String filePath, String shareSnapshot) {
+        this.shareName = shareName;
+        this.filePath = filePath;
+        this.shareSnapshot = shareSnapshot;
+        this.azureFileStorageClient = new AzureFileStorageBuilder().pipeline(azureFileStorageClient.httpPipeline())
+                            .url(azureFileStorageClient.url())
+                            .version(azureFileStorageClient.version())
+                            .build();
+    }
+
+    /**
+     * Constructor of FileAsyncClient
+     * @param endpoint
+     * @param httpPipeline
+     * @param shareName
+     * @param filePath
+     */
+    FileAsyncClient(URL endpoint, HttpPipeline httpPipeline, String shareName, String filePath, String shareSnapshot) {
+        this.shareName = shareName;
+        this.filePath = filePath;
+        this.shareSnapshot = shareSnapshot;
+        this.azureFileStorageClient = new AzureFileStorageBuilder().pipeline(httpPipeline)
+                          .url(endpoint.toString())
+                          .build();
+    }
+
+    /**
+     * Get URL from azureFileStorageClient.
+     * @return
+     */
+    public URL url() throws Exception {
```

Comment:
We shouldn't throw the base exception, we know this should only be a MalformedUrlException